vcs/src/main/java/org/openjdk/skara/vcs/git/GitRepository.java

```diff
@@ -295,6 +295,21 @@ public void clean() throws IOException {
         }
     }
 
+    @Override
+    public void reset(Hash target, boolean hard) throws IOException {
+        var cmd = new ArrayList<String>();
+        cmd.addAll(List.of("git", "reset"));
+        if (hard) {
+           cmd.add("--hard");
+        }
+        cmd.add(target.hex());
+
+        try (var p = capture(cmd.toArray(String[]::new))) {
```

Comment:
There is no need to do `cmd.toArray`, `capture` has an overload for `List<String>`